crates/oxc_linter/src/rules/unicorn/prefer_set_has.rs

```diff
@@ -0,0 +1,900 @@
+            };
+            if call_expression.arguments.len() != 1 || call_expression.optional {
+                return true;
+            }
+            let arg = &call_expression.arguments[0];
+            if arg.is_spread() {
+                return true;
+            }
+            let AstKind::MemberExpression(member_expr) = parent.kind() else {
+                return true;
+            };
+            !is_includes_call(member_expr)
+        }) {
+            return;
+        }
+
+        ctx.diagnostic_with_fix(prefer_set_has_diagnostic(declarator.span), |fixer| {
+            let fixer = fixer.for_multifix();
+            let mut declaration_fix = fixer.new_fix_with_capacity(2);
+            match init {
+                Expression::ArrayExpression(init_node) => {
+                    declaration_fix.push(fixer.insert_text_before(&init_node.span, "new Set("));
+                    declaration_fix.push(fixer.insert_text_after(&init_node.span, ")"));
+                }
+                Expression::CallExpression(call_expr) => {
+                    if call_expr.callee.is_identifier_reference() {
+                        let start = call_expr.span.start;
+                        let end = start + 6;
+                        let span = Span::new(start, end);
+                        declaration_fix.push(fixer.replace(span, "new Set(["));
+                        let start = call_expr.span.end - 1;
+                        let end = start + 1;
+                        let span = Span::new(start, end);
+                        declaration_fix.push(fixer.replace(span, "])"));
+                    } else {
+                        declaration_fix.push(fixer.insert_text_before(&call_expr.span, "new Set("));
+                        declaration_fix.push(fixer.insert_text_after(&call_expr.span, ")"));
+                    }
+                }
+                Expression::NewExpression(new_expr) => {
+                    let start = new_expr.span.start + 4;
+                    let end = start + 6;
+                    let span = Span::new(new_expr.span.start + 4, end);
+                    declaration_fix.push(fixer.replace(span, "Set(["));
+                    let start = new_expr.span.end - 1;
+                    let end = start + 1;
+                    let span = Span::new(start, end);
+                    declaration_fix.push(fixer.replace(span, "])"));
+                }
+                _ => {}
+            }
+
+            let mut references_fix = fixer.new_fix_with_capacity(len);
+            let references = symbol_table.get_resolved_references(symbol_id);
+            for reference in references {
+                let node = ctx.nodes().get_node(reference.node_id());
+                let Some(parent) = ctx.nodes().parent_node(node.id()) else {
+                    continue;
+                };
+                let AstKind::MemberExpression(member_expr) = parent.kind() else {
+                    continue;
+                };
```

Comment:
I don't think we need the actual node here, we just need its ID? We can also use `parent_kind` directly:

```suggestion
                let Some(AstKind::MemberExpression(member_expr)) = ctx.nodes().parent_node(reference.node_id()) else {
                    continue;
                };
```

Reply:
done